apps/hash-api/src/integrations/linear/webhook.ts

```diff
@@ -54,35 +66,102 @@ export const linearWebhook: RequestHandler<{}, string, string> = async (
     );
   }
 
+  const organizationId = payload.organizationId;
+
+  if (!payload.data) {
+    res
+      .status(400)
+      .send(
+        `No data sent with ${payload.action} ${payload.type} webhook payload`,
+      );
+    return;
+  }
+
+  const linearId = payload.data?.id;
+
+  if (!linearId) {
+    res
+      .status(400)
+      .send(`No ID found in ${payload.action} ${payload.type} webhook payload`);
+    return;
+  }
+
+  const { graphApi, vaultClient } = req.context;
+
+  if (!vaultClient) {
+    return;
+  }
+
+  const linearIntegrations = await getAllLinearIntegrationsWithLinearOrgId(
+    { graphApi },
+    { actorId: systemAccountId },
+    { linearOrgId: organizationId },
+  );
+
   if (
     tupleIncludes(["create", "update"], payload.action) &&
-    tupleIncludes(["Issue", "User"], payload.type)
+    tupleIncludes(supportedLinearTypes, payload.type)
   ) {
-    if (!payload.data) {
-      res
-        .status(400)
-        .send(
-          `No data sent with ${payload.action} ${payload.type} webhook payload`,
-        );
-      return;
-    }
-
-    const workflow =
-      `${payload.action}Hash${payload.type}` as const satisfies keyof WorkflowTypeMap;
+    const payloadAction = payload.action;
+    const linearType = payload.type;
 
-    await temporalClient.workflow.start<WorkflowTypeMap[typeof workflow]>(
-      workflow,
-      {
-        taskQueue: "integration",
-        args: [
+    await Promise.all(
+      linearIntegrations.map(async (linearIntegration) => {
+        const syncedWorkspaces = await getSyncedWorkspacesForLinearIntegration(
+          { graphApi },
+          { actorId: systemAccountId },
           {
-            authentication: { actorId: systemAccountId },
-            ownedById: systemAccountId as OwnedById,
-            payload: payload.data,
+            linearIntegrationEntityId:
+              linearIntegration.entity.metadata.recordId.entityId,
           },
-        ],
-        workflowId: `${workflow}-${genId()}`,
-      },
+        );
+
+        await Promise.all(
+          syncedWorkspaces.map(async (workspace) => {
+            /** @todo: only sync with correct teams */
```

Comment:
Is there a ticket for this?

Reply:
Added on in https://github.com/hashintel/hash/pull/3636/commits/d724e4c3834933acf847a5008201a292480ab81a

Reply:
Added on in https://github.com/hashintel/hash/pull/3636/commits/d724e4c3834933acf847a5008201a292480ab81a